### PR TITLE
run,cleanup: document `--olm` mode

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
* doc/user/olm-catalog: add olm-cli.md to explain 'run --olm' and 'cleanup --olm' command usage, and rename 'bundle' doc file to
bundle-cli.md
* cmd/operator-sdk/run: update long-form command description
* doc/user-guide.md: add reference to 'run --olm' doc
* doc,website: regenerate CLI docs

**Motivation for the change:** Comprehensive documentation for `operator-sdk run/cleanup --olm` does not exist.

/kind documentation

### CODE REVIEW

File: website/content/en/docs/cli/operator-sdk_run.md [hunk 1]
```diff
@@ -4,7 +4,14 @@ Run an Operator in a variety of environments
...
 
-Run an Operator in a variety of environments
+This command will run or deploy your Operator in two different modes: locally
+and using OLM. These modes are controlled by setting --local and --olm run mode
+flags. Each run mode has a separate set of flags that configure 'run' for that
+mode. Run 'operator-sdk run --help' for more information on these flags.
+
+Read more about the --olm run mode and configuration options here:
+https://github.com/operator-framework/operator-sdk/blob/master/doc/user/olm-catalog/olm-cli.md
+
```
Comment (camilamacedo86) [inline website/content/en/docs/cli/operator-sdk_run.md]: should not the other changes in the docs be copied to the website as well?

Comment (estroz) [reply website/content/en/docs/cli/operator-sdk_run.md]: Done

File: doc/user-guide.md [hunk 2]
```diff
@@ -574,11 +577,11 @@ func serveCRMetrics(cfg *rest.Config) error {
...
 
-In this scenario, this error may occur because your Operator RBAC roles do not include permissions to LIST the third party API schemas or the schemas which are required to them and will be added with. See that the default SDK implementation will just add the Kubernetes schemas and they will be ignored in the metrics It means that you might need to do an similar implementation to filter the third party API schemas and their dependencies added in order to provide a filtered a List of GVK(GroupVersionKind) to the  `GenerateAndServeCRMetrics` method. 
+In this scenario, this error may occur because your Operator RBAC roles do not include permissions to LIST the third party API schemas or the schemas which are required to them and will be added with. See that the default SDK implementation will just add the Kubernetes schemas and they will be ignored in the metrics It means that you might need to do an similar implementation to filter the third party API schemas and their dependencies added in order to provide a filtered a List of GVK(GroupVersionKind) to the  `GenerateAndServeCRMetrics` method.
```
Comment (jmccormick2001) [inline doc/user-guide.md]: "...will be ignored in the metrics.  It means that you..."

Comment (estroz) [reply doc/user-guide.md]: This PR doesn't change anything in the user guide other than adding a header for OLM deployment. My editor just removes whitespaces. I'll submit a follow-up with the suggested fix.

Comment (estroz) [reply doc/user-guide.md]: PR: #2736

Comment (hasbro17) [reply doc/user-guide.md]: `you might need to do an similar implementation`==> `you might need to do a similar implementation`

Comment (hasbro17) [reply doc/user-guide.md]: @estroz Sorry ignore that. You can add that to the follow up PR for fixes.

**Pull request merged**